ibm/resource_ibm_pi_instance.go

```diff
@@ -125,6 +126,44 @@ func resourceIBMPIInstance() *schema.Resource {
 				Computed:    true,
 				Description: "Storage type for server deployment",
 			},
+			PIInstanceStoragePool: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Storage Pool for server deployment; if provided then pi_affinity_policy and pi_storage_type will be ignored",
+			},
+			PIAffinityPolicy: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Affinity policy for pvm instance being created; ignored if pi_storage_pool provided; for policy affinity requires one of pi_affinity_instance or pi_affinity_volume to be specified; for policy anti-affinity requires one of pi_anti_affinity_instances or pi_anti_affinity_volumes to be specified",
+				ValidateFunc: validateAllowedStringValue([]string{"affinity", "anti-affinity"}),
+			},
+			PIAffinityVolume: {
+				Type:          schema.TypeString,
+				Optional:      true,
```

Comment:
Can all these condition can be achieved RequiredWith

RequiredWith:  []string{"affinity"},...some thing like this for all these arguments

Reply:
It depends on the dynamic value of `pi_affinity_policy` hence we cannot validate it within resource. Also, the API handles the cases very well.